numpy/core/src/umath/loops_comparison.dispatch.c.src

```diff
@@ -0,0 +1,421 @@
+    npyv_lanetype_u8 *dst      = (npyv_lanetype_u8 *) args[2];
+    const npyv_@sfx@ b         = npyv_setall_@sfx@(scalar);
+    const npyv_u8 truemask     = npyv_setall_u8(0x1);
+    const int vstep            = npyv_nlanes_u8;
+
+    for (; len >= vstep; len -= vstep, src += vstep, dst += vstep) {
+#if @len@ >= 8
+        npyv_@sfx@  a1 = npyv_load_@sfx@(src + npyv_nlanes_@sfx@ * 0);
+        npyv_b@len@ c1 = npyv_@VOP@_@sfx@(a1, b);
+#if @len@ >= 16
+        npyv_@sfx@  a2 = npyv_load_@sfx@(src + npyv_nlanes_@sfx@ * 1);
+        npyv_b@len@ c2 = npyv_@VOP@_@sfx@(a2, b);
+#if @len@ >= 32
+        npyv_@sfx@  a3 = npyv_load_@sfx@(src + npyv_nlanes_@sfx@ * 2);
+        npyv_@sfx@  a4 = npyv_load_@sfx@(src + npyv_nlanes_@sfx@ * 3);
+        npyv_b@len@ c3 = npyv_@VOP@_@sfx@(a3, b);
+        npyv_b@len@ c4 = npyv_@VOP@_@sfx@(a4, b);
+#if @len@ == 64
+        npyv_@sfx@  a5 = npyv_load_@sfx@(src + npyv_nlanes_@sfx@ * 4);
+        npyv_@sfx@  a6 = npyv_load_@sfx@(src + npyv_nlanes_@sfx@ * 5);
+        npyv_@sfx@  a7 = npyv_load_@sfx@(src + npyv_nlanes_@sfx@ * 6);
+        npyv_@sfx@  a8 = npyv_load_@sfx@(src + npyv_nlanes_@sfx@ * 7);
+        npyv_b@len@ c5 = npyv_@VOP@_@sfx@(a5, b);
+        npyv_b@len@ c6 = npyv_@VOP@_@sfx@(a6, b);
+        npyv_b@len@ c7 = npyv_@VOP@_@sfx@(a7, b);
+        npyv_b@len@ c8 = npyv_@VOP@_@sfx@(a8, b);
+#endif // @len@ >= 64
+#endif // @len@ >= 32
+#endif // @len@ >= 16
+#endif // @len@ >= 8
+
+#if @len@ == 8
+        npyv_u8 r = npyv_cvt_u8_b8(c1);
+#elif @len@ == 16
+        npyv_u8 r = npyv_cvt_u8_b8(npyv_pack_b8_b16(c1, c2));
+#elif @len@ == 32
+        npyv_u8 r = npyv_cvt_u8_b8(npyv_pack_b8_b32(c1, c2, c3, c4));
+#elif @len@ == 64
+        npyv_u8 r =
+            npyv_cvt_u8_b8(npyv_pack_b8_b64(c1, c2, c3, c4, c5, c6, c7, c8));
+#endif
+        npyv_store_u8(dst, npyv_and_u8(r, truemask));
+    }
+
+    for (; len > 0; --len, ++src, ++dst) {
+        const npyv_lanetype_@sfx@ a = *src;
+        *dst = a @OP@ scalar;
+    }
+}
+#endif
+
+/**end repeat1**/
+/**end repeat**/
+
+/**begin repeat
+ * #kind = equal, not_equal, less, less_equal, greater, greater_equal#
+ * #OP = ==, !=, <, <=, >, >=#
+ * #VOP = cmpeq, cmpneq, cmplt, cmple, cmpgt, cmpge#
+ */
+
+#if NPY_SIMD
+static void simd_binary_@kind@_b8(char **args, npy_intp len)
+{
+    npyv_lanetype_u8 *src1 = (npyv_lanetype_u8 *) args[0];
+    npyv_lanetype_u8 *src2 = (npyv_lanetype_u8 *) args[1];
+    npyv_lanetype_u8 *dst  = (npyv_lanetype_u8 *) args[2];
+    const npyv_u8 truemask = npyv_setall_u8(0x1);
+    const npyv_u8 vzero    = npyv_setall_u8(0x0);
+    const int vstep        = npyv_nlanes_u8;
+
+    for (; len >= vstep;
+         len -= vstep, src1 += vstep, src2 += vstep, dst += vstep) {
+        // Whatever element in src != 0x0 is converted to 0xFF
+        npyv_b8 a = npyv_cmpneq_u8(npyv_load_u8(src1), vzero);
+        npyv_b8 b = npyv_cmpneq_u8(npyv_load_u8(src2), vzero);
+        npyv_b8 c = npyv_@VOP@_u8(npyv_cvt_u8_b8(a), npyv_cvt_u8_b8(b));
```

Comment:
```suggestion
        npyv_b8 c = npyv_@VOP@_b8(a, b);
```
logical bitwise should be used instead, faster in general and its necessary for AVX512 to avoid the plenty of mask conversion.  e.g. `==` -> `and`, `!=` `xor`, `<` => `!a & b`(it would better to implement intrinsic for `andc` ), `<=` `!a | b` same here but for `orc`

Reply:
Done.

See below the mapping (cmp ==> logical bitwise)
`==` ==> `xnor`
`!=` ==>   `xor`
`<` ==> `~a & b  (andc)`
`<=` ==> `~a | b (orc)`
`>` ==> `~b & a  (reverse andc)`
`>=` ==> `~b | a (reverse orc)`

I introduced three new universal intrinsics, as listed below:
- `npyv_andc_b8`
- `npyv_orc_b8`
- `npyv_xnor_b8`

Well, as I am using only for `b8` I created a version only for this datatype. Should I create the same intrinsics for the other dtypes too?

See below the improvements compared to the first commit of this PR.

SSE
```
-     6.78±0.03μs      5.69±0.05μs     0.84  bench_ufunc.CustomComparison.time_less_than_scalar2(<class 'numpy.bool_'>)
-     6.57±0.01μs      5.10±0.01μs     0.78  bench_ufunc.CustomComparison.time_less_than_binary(<class 'numpy.bool_'>)
```

AVX2
```
-     5.21±0.01μs      4.68±0.02μs     0.90  bench_ufunc.CustomComparison.time_less_than_binary(<class 'numpy.bool_'>)
-     5.30±0.02μs      4.65±0.03μs     0.88  bench_ufunc.CustomComparison.time_less_than_scalar2(<class 'numpy.bool_'>)
```


AVX512F
```
-      5.12±0.2μs       4.26±0.1μs     0.83  bench_ufunc.CustomComparison.time_less_than_scalar2(<class 'numpy.bool_'>)
-      5.26±0.2μs       4.34±0.1μs     0.83  bench_ufunc.CustomComparison.time_less_than_scalar1(<class 'numpy.bool_'>)
-      5.49±0.1μs       4.45±0.1μs     0.81  bench_ufunc.CustomComparison.time_less_than_binary(<class 'numpy.bool_'>)
```


AVX512BW
```
-      5.08±0.2μs       4.36±0.1μs     0.86  bench_ufunc.CustomComparison.time_less_than_scalar1(<class 'numpy.bool_'>)
-      5.19±0.2μs       4.34±0.1μs     0.84  bench_ufunc.CustomComparison.time_less_than_scalar2(<class 'numpy.bool_'>)
-      5.45±0.1μs       4.46±0.1μs     0.82  bench_ufunc.CustomComparison.time_less_than_binary(<class 'numpy.bool_'>)
```


VSX
```
-     6.69±0.01μs      5.91±0.02μs     0.88  bench_ufunc.CustomComparison.time_less_than_scalar2(<class 'numpy.bool_'>)
-     6.69±0.01μs      5.90±0.01μs     0.88  bench_ufunc.CustomComparison.time_less_than_scalar1(<class 'numpy.bool_'>)
```

Thank you!

Reply:
> Well, as I am using only for b8 I created a version only for this datatype. Should I create the same intrinsics for the other dtypes too?

not necessary, but we may need to add `andc` for u8 so we can use `cmpeq` instead of `cmpneq` when to comes BOOL data type.

Reply:
Done! I added `andc` for `u8`.